src/ray/gcs/gcs_server/gcs_worker_manager.cc

```diff
@@ -38,44 +38,24 @@ void GcsWorkerManager::HandleReportWorkerFailure(
   worker_failure_data->CopyFrom(request.worker_failure());
   worker_failure_data->set_is_alive(false);
 
-  // Before handle ReportWorkerFailureRequest, you should check if the worker is exists.
-  auto on_get_done = [this, worker_address, worker_id, node_id, worker_failure_data,
-                      reply, send_reply_callback](
-                         const Status &status,
-                         const boost::optional<WorkerTableData> &result) {
-    if (result) {
-      auto on_put_done = [this, worker_address, worker_id, node_id, worker_failure_data,
-                          reply, send_reply_callback](const Status &status) {
-        if (!status.ok()) {
-          RAY_LOG(ERROR) << "Failed to report worker failure, worker id = " << worker_id
-                         << ", node id = " << node_id
-                         << ", address = " << worker_address.ip_address();
-        } else {
-          RAY_CHECK_OK(gcs_pub_sub_->Publish(WORKER_CHANNEL, worker_id.Binary(),
-                                             worker_failure_data->SerializeAsString(),
-                                             nullptr));
-        }
-        GCS_RPC_SEND_REPLY(send_reply_callback, reply, status);
-      };
-
-      // The worker exists in worker table, you can update the info of this worker.
-      Status report_status = gcs_table_storage_->WorkerTable().Put(
-          worker_id, *worker_failure_data, on_put_done);
-      if (!report_status.ok()) {
-        on_put_done(report_status);
-      }
+  auto on_done = [this, worker_address, worker_id, node_id, worker_failure_data, reply,
+                  send_reply_callback](const Status &status) {
+    if (!status.ok()) {
+      RAY_LOG(ERROR) << "Failed to report worker failure, worker id = " << worker_id
+                     << ", node id = " << node_id
+                     << ", address = " << worker_address.ip_address();
     } else {
-      // The worker doesn't exists in worker table.
-      RAY_LOG(WARNING) << "Failed to report worker failure, the worker doesn't "
-                          "exist, worker id = "
-                       << worker_id << ", node id = " << node_id
-                       << ", address = " << worker_address.ip_address();
-      GCS_RPC_SEND_REPLY(send_reply_callback, reply, status);
+      RAY_CHECK_OK(gcs_pub_sub_->Publish(WORKER_CHANNEL, worker_id.Binary(),
+                                         worker_failure_data->SerializeAsString(),
+                                         nullptr));
     }
+    GCS_RPC_SEND_REPLY(send_reply_callback, reply, status);
   };
-  Status status = gcs_table_storage_->WorkerTable().Get(worker_id, on_get_done);
+
+  Status status =
```

Comment:
Can you write a comment about this PR? That could be potentially useful if any issue happens in the future.

Reply:
Good idea, thanks